modules/contact/src/actions/ContactAction.C

```diff
@@ -336,19 +357,25 @@ ContactAction::addMortarContact()
     // Add the lagrange multiplier on the secondary subdomain.
     const auto addLagrangeMultiplier =
         [this, &secondary_subdomain_name, &displacements](const std::string & variable_name,
-                                                          const int codimension,
+                                                          const int min_lm_order,
+                                                          const bool frictional,
                                                           const Real scaling_factor) //
     {
       InputParameters params = _factory.getValidParams("MooseVariableBase");
-      params.set<bool>("use_dual") = _use_dual;
+
+      // Allow the user to select "weighted" constraints and standard bases (use_dual = false)
+      // Unless it's for testing purposes, this combination isn't recommended
+      params.set<bool>("use_dual") = (_mortar_approach == MortarApproach::Weighted && _use_dual);
 
       mooseAssert(_problem->systemBaseNonlinear().hasVariable(displacements[0]),
                   "Displacement variable is missing");
       const auto primal_type =
           _problem->systemBaseNonlinear().system().variable_type(displacements[0]);
-      const int lm_order = std::max(primal_type.order.get_order() - codimension, 0);
+      const int codimension = frictional && !(_mortar_approach == MortarApproach::Weighted);
+      // Order of LM is independent of whether it enforces normal contact or frictional contact.
+      const int lm_order = std::max(primal_type.order.get_order() - codimension, min_lm_order);
 
-      if (primal_type.family == MONOMIAL || (primal_type.family == LAGRANGE && lm_order < 1))
+      if ((primal_type.family == LAGRANGE) && (lm_order < 1))
```

Comment:
```suggestion
      if (primal_type.family == LAGRANGE && lm_order < 1)
```

if you're changing stuff anyways